File: fix_icons.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'frontend', 'src', 'components', 'ChartPane.jsx');

// Read file as buffer to preserve encoding
let content = fs.readFileSync(filePath, 'utf8');

// Replace using Unicode escape sequences
content = content.replace(/\u0440\u045F\u2019\u0403/g, '\u25CF');  // ● filled circle
content = content.replace(/\u0440\u045F\u2019\u0403\u201C\u0440\u045F\u2014\u0401/g, '\u25CB');  // ○ empty circle
content = content.replace(/\u0432\u045A\u2122/g, '\u2699');  // ⚙ gear
content = content.replace(/\u0432\u2020\u2019/g, '\u25B2');  // ▲ up triangle
content = content.replace(/\u0432\u2020\u201C/g, '\u25BC');  // ▼ down triangle

// Write back
fs.writeFileSync(filePath, content, 'utf8');

console.log('Icons fixed!');

File: fix_icons.py
# -*- coding: utf-8 -*-
import codecs
import re

file_path = r"c:\Users\User\Desktop\Project's Antigravity\Work\AlgoResearch Lab\frontend\src\components\ChartPane.jsx"

# Read file
with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
    content = f.read()

# Replace broken patterns with proper symbols using regex
# Pattern for broken eye icons
content = re.sub(r"рџ'Ѓ", "\u25CF", content)  # ● filled circle
content = re.sub(r"рџ'ЃвЂЌрџ—Ё", "\u25CB", content)  # ○ empty circle

# Pattern for broken gear
content = re.sub(r"вљ™", "\u2699", content)  # ⚙ gear

# Pattern for broken arrows
content = re.sub(r"в†'", "\u25B2", content)  # ▲ up triangle
content = re.sub(r"в†"", "\u25BC", content)  # ▼ down triangle

# Write back with UTF-8 BOM
with open(file_path, 'w', encoding='utf-8-sig') as f:
    f.write(content)

print("Icons fixed successfully!")
